ci-operator/step-registry/rosa/sts/cluster/provision/rosa-sts-cluster-provision-commands.sh

```diff
@@ -182,12 +182,26 @@ PROXY_SWITCH=""
 if [[ "$ENABLE_PROXY" == "true" ]]; then
   # Get the proxy information from the previous steps, and replace the value here
   proxy_private_url=$(< "${SHARED_DIR}/proxy_private_url")
-  TRUST_BUNDLE_FILE="bundle_file"
-  if [[ -z "${proxy_private_url}" ]] || [[ -z "${TRUST_BUNDLE_FILE}" ]]; then
-    echo -e "The http_proxy, the http_proxy and the additional_trust_bundle_file are mandatory."
+  TRUST_BUNDLE_FILE="${SHARED_DIR}/bundle_file"
+  if [[ -z "${proxy_private_url}" ]]; then
+    echo -e "The http_proxy, and https_proxy URLs are mandatory when specifying use of proxy."
     exit 1
   fi
-  PROXY_SWITCH="--http-proxy ${proxy_private_url} --https-proxy ${proxy_private_url} --additional-trust-bundle-file ${TRUST_BUNDLE_FILE}"
+  if [[ -f "${TRUST_BUNDLE_FILE}" ]]; then
+    echo -e "Using proxy with requested additional trust bundle"
+    PROXY_SWITCH="--http-proxy ${proxy_private_url} --https-proxy ${proxy_private_url} --additional-trust-bundle-file ${TRUST_BUNDLE_FILE}"
+  else
+    echo -e "Using proxy with no additional trust bundle provided"
+    PROXY_SWITCH="--http-proxy ${proxy_private_url} --https-proxy ${proxy_private_url}"
+  fi
+  PUBLIC_SUBNET_IDs=$(cat ${SHARED_DIR}/public_subnet_ids | tr -d "[']")
+  PRIVATE_SUBNET_IDs=$(cat ${SHARED_DIR}/private_subnet_ids | tr -d "[']")
+  # fixme, list of subnets returned only work for multi az cluster need to remove subnets for single az
+  if [[ -z "${PUBLIC_SUBNET_IDs}" ]] || [[ -z "${PRIVATE_SUBNET_IDs}" ]]; then
+    echo -e "The public_subnet_ids and the the private_subnet_ids are mandatory when using the proxy."
+    exit 1
+  fi
+  SUBNET_ID_SWITCH="--subnet-ids ${PUBLIC_SUBNET_IDs},${PRIVATE_SUBNET_IDs}"
```

Comment:
`SUBNET_ID_SWITCH` is not used only for proxy but also for the other options, etc. HOSTED_CP=true. So let us import a new variable, etc, ENABLE_SUBNET. If ENABLE_SUBNET=true, we will set SUBNET_ID_SWITCH.

Reply:
ok I believe we also require subnets for private clusters as well, otherwise providing your own subnet seems to be an option for the other cluster types.

Reply:
Yes, private cluster requires subnets too.

Reply:
added ENABLE_BYOVPC to cover cases where byo vpc is optional, I didn't want to make it required for hosted cp, private and proxy as I didn't want to break existing workflows.

Reply:
As we have had `ENABLE_BYOVPC` block to set `SUBNET_ID_SWITCH`, we need to remove these from `ENABLE_PROXY`.